features/admin.actions.v1/models/actions.ts

```diff
@@ -0,0 +1,343 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Different Authentication Types.
+ */
+export enum AuthenticationType {
+    NONE = "NONE",
+    BASIC = "BASIC",
+    API_KEY = "API_KEY",
+    BEARER = "BEARER",
+}
+
+/**
+ * Different Action Types.
+ */
+export enum ActionType {
+    PRE_ISSUE_ACCESS_TOKEN = "PRE_ISSUE_ACCESS_TOKEN",
+    PRE_UPDATE_PASSWORD = "PRE_UPDATE_PASSWORD",
+    PRE_UPDATE_PROFILE = "PRE_UPDATE_PROFILE",
+    PRE_REGISTRATION = "PRE_REGISTRATION"
+}
+
+/**
+ * Action Status.
+ */
+enum ActionStatus {
+    ACTIVE,
+    INACTIVE
+}
+
+/**
+ * Interface for the authentication type dropdown options.
+ */
+export interface AuthTypeDropdownOption {
+    key: AuthenticationType;
+    text: string;
+    value: AuthenticationType;
+}
+
+/**
+ *  Action configuration.
+ */
+export interface ActionInterface {
+    /**
+     * Name of the Action.
+     */
+    name: string;
+    /**
+     * Description of the Action.
+     */
+    description?: string;
+    /**
+     * Endpoint configuration of the Action.
+     */
+    endpoint: EndpointInterface;
+}
+
+/**
+ *  Endpoint configuration.
+ */
+export interface EndpointInterface {
+    /**
+     * External endpoint.
+     */
+    uri: string;
+    /**
+     * Authentication configurations of the Action.
+     */
+    authentication: AuthenticationInterface;
+}
+
+/**
+ *  Endpoint authentication configuration.
+ */
+export interface AuthenticationInterface {
+    /**
+     * Authentication Type.
+     */
+    type: AuthenticationType;
+    /**
+     * Authentication properties.
+     */
+    properties: AuthPropertiesInterface;
```

Comment:
use Pick and define interfaces for authtypes

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/6722/commits/d96314b591f6f6f33b709b0600c399b789e7fcef